src/dowel/csv_output.py

```diff
@@ -42,12 +42,27 @@ def record(self, data, prefix=''):
                 self._writer.writeheader()
 
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+                                
+                # update new field name
+                new_field = set(to_csv.keys())
+                self._fieldnames.update(new_field)
+
+                # read existing file and write back with new fieldname
+                # self._log_file contains: name, mode, encoding
+                # <_io.TextIOWrapper name='out.csv' mode='w' encoding='cp1252'>
+                with open(self._log_file.name, "r", newline="") as csvfileog:
+                    og_data = csv.DictReader(csvfileog)
```

Comment:
`og_data` might be a little unclear to some people, `original_data` would probably be more clear :)

Reply:
thanks for pointing out, I used to write homework in this way, changes has been made for the project